### ISSUE TITLE
Allow regexes in `t.like`

### ISSUE THREAD

**Issue by tommy-mitchell** [avajs/ava#3309]
The `message` property in `t.throws` is really ergonomic - it can be a `string`, `RegExp`, or even a function. `t.like` should be just as ergonomic with strings:

```js
const output = foo();

t.like(output, {
  name: 'foo',
  source: /foo comes from/,
});
```

---

A motivating example is trying to convert [this test](https://github.com/sindresorhus/npm-user/blob/0ea3f07c8b70023fdd8de39bb5ad894ef67688d4/test.js#L4-L9) from `npm-user` to use `t.like`:

```js
const user = await npmUser('sindresorhus');

t.like(user, {
	name: 'Sindre Sorhus',
	avatar: /npm-avatar/,
	email: 'sindresorhus@gmail.com',
});
```

**Comment by novemberborn** [comment]
Currently `like` is a partial `deepEqual`, selecting from the `actual` (first argument) just those properties that exist in the `expected` (second argument). How would we do that if `expected` contains a regular expression value?

**Comment by Saberian1992** [comment]
> Currently `like` is a partial `deepEqual`, selecting from the `actual` (first argument) just those properties that exist in the `expected` (second argument). How would we do that if `expected` contains a regular expression value?

**Comment by tommy-mitchell** [comment]
Maybe:

- If `expected.prop` is a `RegExp` and `actual.prop` is a `string`:

  1. Check if `actual.prop` matches `expected.prop`
  
  2. If `false`, fail the test. The current descriptors (string vs regex) can stay as-is:
    ```diff
    - str: 'string',
    + str: /string/,
    ```

- If `expected.prop` is a `RegExp` and `actual.prop` is a `RegExp`:
  1. No change. Only pass if the two regexes are the same.

- If `expected.prop` is a `string` and `actual.prop` is a `RegExp`:
  1. No change. The test should still fail.

**Comment by novemberborn** [comment]
I agree that this looks useful, and your logic makes sense, but I don't immediately see how this could be implemented. Don't let that stop you though 😉 

It may also technically be a breaking change, in that if you've written an existing assertion expecting a regular expression, and your actual value changes to a string, that happens to match… the test passes when it shouldn't. OK that seems unlikely.
